src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -8,16 +9,21 @@
     /// </summary>
     public abstract class BehaviorContext : ContextBag
     {
+        //TODO DanielTim: Does it make to force us to provide a RootContext here (to ensure Settings are set)? RootContext currently is private.
         /// <summary>
         /// Create an instance of <see cref="BehaviorContext"/>.
         /// </summary>
         /// <param name="parentContext">The parent context.</param>
-        protected BehaviorContext(ContextBag parentContext) : base(parentContext)
+        protected BehaviorContext(BehaviorContext parentContext) : base(parentContext)
         {
-          
         }
 
         /// <summary>
+        /// The readonly settings.
+        /// </summary>
+        public ReadOnlySettings Settings => Get<ReadOnlySettings>();
```

Comment:
Can we just add a the mapper to a `pipeline.Sevices` thingy?


featureContext.Pipeline.AddService(mapper)

and then at runtime

context.Services.Get<IMessgeMapper>()?

Is this gets to big perhaps move this to a seprate pull?